File: backend/app/schemas/system.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class HealthResponse(BaseModel):
    status: Literal["ok"]
    service: Literal["backend"]


class ReadinessChecks(BaseModel):
    database: Literal["ok"]
    redis: Literal["ok"]


class ReadinessResponse(BaseModel):
    status: Literal["ready"]
    checks: ReadinessChecks


class SystemInfoResponse(BaseModel):
    application: str
    environment: str
    backend_version: str
    debug: bool
    server_time: datetime
    database_schema_version: str

File: frontend/src/types/system.ts
export type CheckStatus = "checking" | "ok" | "unavailable";

export interface ReadinessResponse {
  status: "ready";
  checks: {
    database: "ok";
    redis: "ok";
  };
}

export interface ApiErrorResponse {
  error?: {
    code?: string;
    message?: string;
    request_id?: string;
    details?: {
      checks?: {
        database?: "ok" | "unavailable";
        redis?: "ok" | "unavailable";
      };
    };
  };
}

export interface SystemStatus {
  backend: CheckStatus;
  database: CheckStatus;
  redis: CheckStatus;
}
